File: boj/2195.py
s=input()
p=input()
# 딱히 뭐가 없음 
start = 0
cnt=0
for i in range(len(p)):
    # contains
    substr = p[start:i+1]
    #print(start, i+1)
    #print(substr)
    if substr in s:
        continue
    else:
        #print("copy")
        start=i
        cnt+=1
cnt+=1
print(cnt)

File: boj/2195.java
import java.io.*;
import java.math.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        String s1 = br.readLine();
        String s2 = br.readLine();

        int start = 0;
        int ans = 0;

        for(int i = 0; i < s2.length(); i++){
            String substr = s2.substring(start, i+1);
            if (!s1.contains(substr)) {
                start=i;
                ans+=1;
            }
        }
        ans+=1;
        System.out.println(ans);
        bw.flush();
        bw.close();
    }
}